scripts/build/arch/x86.sh

```diff
@@ -20,4 +20,49 @@ CT_DoArchTupleValues() {
         esac
     fi
     CT_TARGET_ARCH="${CT_TARGET_ARCH}${CT_ARCH_SUFFIX}"
+
+    # Shouldn't be possible to specify this (CT_TARGET_SYS is not specified by the user,
+    # it is computed by scripts/functions from libc choices). But trap if such invalid
+    # values ever come from the caller:
+    case "${CT_TARGET_ARCH}-${CT_TARGET_SYS}" in
+        i[34567]86-gnux32)
+            CT_DoLog ERROR "Invalid CT_TARGET: i[34567]86-<vendor>-<os>-gnux32 is invalid."
+            CT_DoLog ERROR "CT_TARGET: ${CT_TARGET}"
+            CT_Abort "Go read: https://wiki.debian.org/Multiarch/Tuples"
+            ;;
+    esac
+}
+
+#------------------------------------------------------------------------------
+# Get multilib architecture-specific target
+# Usage: CT_DoArchMultilibTarget "multilib flags" "target tuple"
+CT_DoArchMultilibTarget ()
+{
+    local multi_flags="${1}"
+    local target="${2}"
+
+    local bit=default
+    local abi=default
+
+    for m in $multi_flags; do
+        case "$m" in
+            -m32)  bit=32; abi=default;;
+            -m64)  bit=64; abi=default;;
+            -mx32) bit=64; abi=x32;;
```

Comment:
To be consistent across all the `CT_DoArchMultilibTarget` functions,
what about:

    local bit32=false
    local bit64=false
    local abi_default=false
    local abi_x32=false

    case "$m" in
        -m32)  bit32=true; abi_default=true;;
        -m64)  bit64=true; abi_default=true;;
        -mx32) bit64=true; abi_x32=true;;
    esac

And use them as their counterparts are used in other
`CT_DoArchMultilibTarget` functions:

    case "${target}" in
        x86_64-*)      $bit32 && target=${target/#x86_64-/i386-} ;;
        i[34567]86-*)  $bit64 && target=${target/#i[34567]86-/x86_64-} ;;
    esac

and so on...